apps/api/src/services/distributed-lock/index.ts

```diff
@@ -0,0 +1,135 @@
+import Redis from 'ioredis';
+import Redlock, { Lock, ResourceLockedError, Settings } from 'redlock';
+import { Logger } from '@nestjs/common';
+
+const LOG_CONTEXT = 'DistributedLock';
+
+const redisUrl = `${process.env.REDIS_HOST}:${process.env.REDIS_PORT}` || 'localhost:6379';
+
+interface ILockOptions {
+  resource: string;
+  ttl: number;
+}
+
+let distributedLock: Redlock;
+let lockCounter = 0;
+
+export const startup = (
+  settings: Settings = {
+    automaticExtensionThreshold: 500,
+    driftFactor: 0.01,
+    retryCount: 50,
+    retryDelay: 100,
+    retryJitter: 200,
+  }
+): void => {
+  if (distributedLock) {
```

Comment:
Could we maybe also check that the appropriate REDIS_HOST env variable exists? And skip the lock functionality so the system can also operate potentially with the locking mechanism as usual?

Reply:
That's a good point. Though it won't work properly, but at least would somehow work.